test/ipfs-factory/daemon-spawner.js

```diff
@@ -35,7 +35,7 @@ function Factory () {
 
       nodes.push(node)
 
-      callback(null, node.apiAddr)
+      callback(null, node.apiAddr.toString())
```

Comment:
Your senses were correct! I made another PR that fixes this: https://github.com/ipfs/js-ipfs-api/pull/525 along with adding a test to test the constructor for both Multiaddr type and a multiaddr string. Tested against ipfsd-ctl@0.19.0 and the tests pass.